core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -60,8 +60,10 @@
     public static final ParseField TYPE_FIELD = new ParseField("type");
     public static final ParseField QUERY_FIELD = new ParseField("query");
 
-    /** The default name for the match query */
-    public static final String NAME = "match";
+    /** The names for the match query. */
+    // TODO replace this with a Types like object
+    public static final Names NAMES = new Names("match", "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix",
```

Comment:
I agree with Luca here the JavaDocs for `Names#primary()` says that its the name outputted by `toXContent()` which really makes it feel like the primary names is the name that we want everyone to use and the other names are supported but effectively deprecated. I think we should fix the few queries this this so they only use one preferred name and the rest and deprecated to be removed in 6.0.